manifests/unit_file.pp

```diff
@@ -1,22 +1,57 @@
-# -- Define: systemd::unit_file
-# Creates a unit file and reloads systemd
+# Creates a systemd unit file
+#
+# @api public
+#
+# @see systemd.unit(5)
+#
+# @attr name [Pattern['^.+\.(service|socket|device|mount|automount|swap|target|path|timer|slice|scope)$']]
+#   The target unit file to create
+#
+#   * Must not contain ``/``
+#
+# @attr path
+#   The main systemd configuration path
+#
+# @attr content
+#   The full content of the unit file
+#
+#   * Mutually exclusive with ``$source``
+#
+# @attr source
+#   The ``File`` resource compatible ``source``
+#
+#   * Mutually exclusive with ``$content``
+#
+# @attr target
+#   If set, will force the file to be a symlink to the given target
+#
+#   * Mutually exclusive with both ``$source`` and ``$content``
+#
 define systemd::unit_file(
-  $ensure = file,
-  $path = '/etc/systemd/system',
-  $content = undef,
-  $source = undef,
-  $target = undef,
+  Enum['file', 'absent']         $ensure  = 'file',
```

Comment:
compared to the service ensure above, we don't allow true/false here, which is good in my opinion.